### PR TITLE
Ng lsn3 exercises

### PR DESCRIPTION
**Pull request by CarlyLanglois**

### CODE REVIEW

File: src/chapters/angular-lsn3/exercises.rst [hunk 1]
```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
...
+handlers and add the following:
+
+.. sourcecode:: TypeScript
+
+   rocketImage.style.bottom = '0px';
+
+New Requirements
 ----------------
+
+You'll notice, a user can move the rocket before the shuttle is officially taken off. One can also
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: Rephrasing suggestion: How about, "Right now, a user can move the rocket before it has officially taken off or abort the mission while the rocket is still on the ground."

File: src/chapters/angular-lsn3/exercises.rst [hunk 2]
```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
...
+Along those same lines, we'll want to modify a couple of our control 
+button handlers to update ``rocketImage``'s position when the status
+changes. Pass in ``rocketImage`` to your *Land* and *Abort Mission* 
+handlers and add the following:
+
+.. sourcecode:: TypeScript
+
+   rocketImage.style.bottom = '0px';
+
+New Requirements
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: These are nicely explained.  I'd only suggest setting them up as another ordered list to give a visual breakdown of the different steps.

File: src/chapters/angular-lsn3/exercises.rst [hunk 3]
```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
...
+.. sourcecode:: html+ng2
+   :linenos:
+      
+   <div class="container-control-buttons">
+      <button (click)="handleTakeOff()" [disabled]="!status.takeOffEnabled">Take Off</button>
+      <button (click)="handleLand(rocketImage)" [disabled]="!status.landEnabled">Land</button>
+      <button (click)="handleMissionAbort(rocketImage)" [disabled]="!status.missionAbortEnabled">Abort Mission</button>
+   </div>
+
+Now, only the *Take Off* button is enabled and the other two controls are disabled on first load,
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: Comma fixes: "Now only the *Take Off* button is enabled, and the other..."

Alternatively: "Now, based on the boolean values we added to ``initStatus``, only the *Take Off* control button is enabled when the rocket is on the ground."

File: src/chapters/angular-lsn3/exercises.rst [hunk 4]
```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
...
+based on the boolean values we added to our ``initStatus`` object.
+
+Add these properties and the appropriate boolean values to the other status variables to toggle
+the enabled/disabled status of the controls.
+
+Lastly, we shouldn't be able to move the rocket if it hasn't taken off. To toggle the status of
+the directions buttons, we could add more properties to our status object. However, we know we only
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: "direction" instead of "directions".

File: src/chapters/angular-lsn3/exercises.rst [hunk 5]
```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
...
+
+   <div *ngIf="!status.takeOffEnabled">
+      <button (click)="handleUpClick(rocketImage)">Up</button>
+      <button (click)="handleDownClick(rocketImage)">Down</button>
+      <button (click)="handleRightClick(rocketImage)">Right</button>
+      <button (click)="handleLeftClick(rocketImage)">Left</button>
+   </div>
+
+
+Bonus Mission
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: Nice choices on the bonus missions.

File: src/chapters/angular-lsn3/exercises.rst [hunk 6]
```diff
@@ -11,31 +12,255 @@ The starter code for the exercises is in the same
...
+-----------------------------------------------
+
+For requirements 1-3, you'll see that the properties that need to be changed 
+have been arranged into several status objects, ``initStatus``, ``shuttleTakeOff``,
+``shuttleLand``, and ``shuttleMissionAbort``. Depending on the state of the rocket,
+one of these status objects will assign the values seen in the app view.
+
+The ``color``, ``height``, and ``message`` properties will all be used as
+template variables to dynamically update the view. In ``app.component.html``,
+change line 19 to the following:
```
[the remark this answers is not in the document]
Comment (CarlyLanglois) [reply src/chapters/angular-lsn3/exercises.rst]: im not sure i understand? is this the same request as above? or are you looking for the starter html file as well?

File: src/chapters/angular-lsn3/exercises.rst [hunk 7]
```diff
@@ -11,31 +12,250 @@ The starter code for the exercises is in the same
...
+
+Now, based on the boolean ``takeOffEnabled``, only the *Take Off* control button 
+is enabled when the rocket is on the ground.
+
+Update the control button click handlers to toggle the enabled/disabled status 
+of the controls using this value.
+
+3. For another improvement, we shouldn't be able to move the rocket if it hasn't taken off. To toggle the status of
+the direction buttons, we could add more boolean checks to our component. However, we know we only
+want these buttons to be accessible when the *Take Off* button is not. We can therefore take advantage
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: Rephrasing suggestion: "We can therefore re-use ``takeOffEnabled`` to determine if the user can click the direction buttons."

File: src/chapters/angular-lsn3/exercises.rst [hunk 8]
```diff
@@ -11,31 +12,250 @@ The starter code for the exercises is in the same
...
+of this property we already defined to determine if the user can click the direction buttons.
+
+.. sourcecode:: html+ng2
+   :linenos:
+      
+   <button (click)="moveRocket(rocketImage, 'up')" [disabled]="takeOffEnabled">Up</button>
+
+In fact, since all four direction buttons share the same requirements for disablement, we can take
+advantage of our old friend ``ngIf`` to display the whole set based on ``takeOffEnabled``.
```
Comment (jimflores5) [inline src/chapters/angular-lsn3/exercises.rst]: I've been including the * in ``*ngIf`` throughout lessons 2 and 3 to emphasize the need for using the asterisk in structural directives.

**Comment by jimflores5** [comment]
Changes approved!

**Pull request merged**